resources/mysql.go

```diff
@@ -0,0 +1,160 @@
+package resources
+
+import (
+	"database/sql"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"time"
+
+	// install mysql driver
+	_ "github.com/go-sql-driver/mysql"
+)
+
+// MysqlInstance is MySQL instance installed by bytebase for testing.
+type MysqlInstance struct {
+	// basedir is the directory where the mysql binary is installed.
+	basedir string
+	// datadir is the directory where the mysql data is stored.
+	datadir string
+	// port is the port of the mysql instance.
+	port int
+	// proc is the process of the mysql instance.
+	proc *os.Process
+}
+
+// Port returns the port of the mysql instance.
+func (i MysqlInstance) Port() int { return i.port }
+
+// Start starts the mysql instance on the given port, outputs to stdout and stderr.
+// Waits at most `waitSec` seconds for the mysql instance to ready for connection.
+// If `waitSec` is 0, it returns immediately.
+func (i *MysqlInstance) Start(port int, stdout, stderr io.Writer, waitSec int) (err error) {
+	i.port = port
+	if i.port == 0 {
```

Comment:
If all existing cases just want to pick a random port, then maybe we just don't need to declare port as a parameter.